### PR TITLE
feat: add `reference_file_schema_uri` to LoadJobConfig, ExternalConfig

### PR DESCRIPTION
**Pull request by aribray**
Current behavior: 
- for load jobs from federated formats like AVRO, PARQUET, and ORC, BigQuery uses the schema of whichever file is lexicographically last.

Example:
```
source_uris = [
    "gs://{project}/{bucket_name}/c-file.avro", 
    "gs://{project}/{bucket_name}/b-file.avro",
    "gs://{project}/{bucket_name}/r-file.avro",
]
```
`"gs://{project}/{bucket_name}/r-file.avro"`  is lexicographically last

New behavior:
- The `reference_file_schema_uri` field allows users to specify the schema
 - The `reference_file_schema_uri` does not have to be a file from the `source_uris` list
 -  To prevent data loss, the `reference_file_schema_uri` should be a superset of the schemas in the `source_uris` list

Googlers see 246809557

### CODE REVIEW

**Review by Neenu1995 — APPROVED** [review]

File: tests/system/test_client.py [hunk 1]
```diff
@@ -1052,6 +1052,211 @@ def test_load_table_from_file_w_explicit_location(self):
...
+        # b-twitter schema: (username, tweet, timestamp)
+        # c-twitter schema: (username, tweet)
+        source_uris = [
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/a-twitter.avro",
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/b-twitter.avro",
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/c-twitter.avro",
+        ]
+
+        # Because referenceFileSchemaUri is set as a-twitter, the table will have a-twitter schema
+        reference_file_schema_uri = "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/a-twitter.avro"
```
Comment (leahecole) [inline tests/system/test_client.py:1081]: non blocking nit)- for maintainability, is it worth setting these to variables or constants and then using the same variable in `source_uris` and `reference_file_schema_uri`?

File: tests/system/test_client.py [hunk 2]
```diff
@@ -1052,6 +1052,211 @@ def test_load_table_from_file_w_explicit_location(self):
...
```
Comment (leahecole) [inline tests/system/test_client.py:1133]: same non blocking nit about variables

Comment (leahecole) [reply tests/system/test_client.py:1133]: tbh, I am torn about this because it might make it slightly less readable? Not sure I want to die on this hill

**Review by leahecole — APPROVED** [review]
Yeah with the nits, I'm honestly torn. Use your best judgment - it's nbd if it's not changed.

File: tests/system/test_client.py (hunk 1 above)
Comment (aribray) [reply tests/system/test_client.py:1081]: Good catch, I think it's worth it

**Review by leahecole — APPROVED** [review]

**Pull request merged**